src/electron/add_tap_device.bat

```diff
@@ -131,3 +106,20 @@ if %errorlevel% neq 0 (
   exit /b 1
 )
 echo TAP network device added and configured successfully
+exit /b 0
+
+:: Waits up to a minute until a device is visible to netsh. Accepts the device name as a parameter.
+:: Exits with a non-zero code if the operation times out.
+:wait_for_device
+echo Testing that the network device "%~1" is visible to netsh...
+netsh interface ip show interfaces | find "%~1" >nul 2>&1
+if %errorlevel% equ 0 exit /b 0
+for /l %%n in (1, 1, 6) do (
+  echo Waiting... %%n
+  :: timeout doesn't like the environment created by nsExec::ExecToStack and exits with:
+  :: "ERROR: Input redirection is not supported, exiting the process immediately."
+  waitfor /t 10 thisisnotarealsignalname >nul 2>&1
+  netsh interface ip show interfaces | find "%~1" >nul 2>&1
+  if %errorlevel% equ 0 exit /b 0
+)
+exit /b 1
```

Comment:
Let's log that we timed out

Reply:
We'll know we timed out from the attempt number. Whenever possible, I have tried to minimize logging verbosity. Relatedly, note that we don't exit if the wait times out; I added comments explaining that we will attempt to configure the device even if we timed out.